middle_end/flambda/flambda.ml

```diff
@@ -1045,13 +1044,20 @@ let create_function_declaration ~params ~body ~stub ~dbg
       "Stubs may not be annotated as [Always_specialise]: %a"
       print body
   end;
+  let dbg_origin =
+    match Closure_origin.debug_info closure_origin with
+    | None ->
+      Misc.fatal_errorf "Debug info missing on closure origin %a"
+        Closure_origin.print closure_origin
+    | Some dbg_origin -> dbg_origin
+  in
```

Comment:
Since it is expected for a closure_origin to always have a debuginfo, I would rather move that check there and change the type of `Closure_origin.debuginfo` to remove the option

Reply:
Closure_id and Closure_origin now have a method `val debug_info : t -> Debuginfo.t`